From 3f52b26e6d701e57e1fe0ba7602c201770ecea82 Mon Sep 17 00:00:00 2001
From: Daniel Larraz <daniel-larraz@users.noreply.github.com>
Date: Fri, 11 Oct 2024 17:43:23 -0500
Subject: [PATCH 119/312] cmake: Require Java 1.8 (#11275)

It also ensures the compiled classes are compatible with the minimum
required version.
---
 src/api/java/CMakeLists.txt | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/api/java/CMakeLists.txt b/src/api/java/CMakeLists.txt
index 34fe18108..64f36293b 100644
--- a/src/api/java/CMakeLists.txt
+++ b/src/api/java/CMakeLists.txt
@@ -102,7 +102,14 @@ add_custom_command(
 add_custom_target(generate-java-types DEPENDS ${JAVA_TYPES_FILES})
 
 # find java
-find_package(Java COMPONENTS Development REQUIRED)
+find_package(Java 1.8 COMPONENTS Development REQUIRED)
+if(Java_VERSION_MAJOR GREATER_EQUAL 9)
+  # Compile option --release is available from JDK 9 onwards.
+  # It ensures the compiled classes are compatible with
+  # the minimum required version. It expects a single number
+  # and does not support 1.X syntax used in JDK < 9.
+  set(CMAKE_JAVA_COMPILE_FLAGS "--release" "8")
+endif()
 include(UseJava)
 
 # specify java source files
-- 
2.47.1

